aws/resource_aws_autoscalingplans_scaling_plan.go

```diff
@@ -0,0 +1,1081 @@
+		// The order of the returned elements is deterministic.
+		for _, v := range vTargetTrackingConfigurations.List() {
+			buf.WriteString(fmt.Sprintf("%d-", autoScalingPlansTargetTrackingConfigurationHash(v)))
+		}
+	}
+
+	return hashcode.String(buf.String())
+}
+
+func autoScalingPlansTargetTrackingConfigurationHash(vTargetTrackingConfiguration interface{}) int {
+	var buf bytes.Buffer
+
+	mTargetTrackingConfiguration := vTargetTrackingConfiguration.(map[string]interface{})
+	if vCustomizedScalingMetricSpecification, ok := mTargetTrackingConfiguration["customized_scaling_metric_specification"].([]interface{}); ok && len(vCustomizedScalingMetricSpecification) > 0 && vCustomizedScalingMetricSpecification[0] != nil {
+		mCustomizedScalingMetricSpecification := vCustomizedScalingMetricSpecification[0].(map[string]interface{})
+		if v, ok := mCustomizedScalingMetricSpecification["dimensions"].(map[string]interface{}); ok {
+			buf.WriteString(fmt.Sprintf("%d-", stableMapHash(v)))
+		}
+		if v, ok := mCustomizedScalingMetricSpecification["metric_name"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+		if v, ok := mCustomizedScalingMetricSpecification["namespace"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+		if v, ok := mCustomizedScalingMetricSpecification["statistic"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+		if v, ok := mCustomizedScalingMetricSpecification["unit"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+	}
+	if v, ok := mTargetTrackingConfiguration["disable_scale_in"].(bool); ok {
+		buf.WriteString(fmt.Sprintf("%t-", v))
+	}
+	if v, ok := mTargetTrackingConfiguration["estimated_instance_warmup"].(int); ok {
+		buf.WriteString(fmt.Sprintf("%d-", v))
+	}
+	if vPredefinedScalingMetricSpecification, ok := mTargetTrackingConfiguration["predefined_scaling_metric_specification"].([]interface{}); ok && len(vPredefinedScalingMetricSpecification) > 0 && vPredefinedScalingMetricSpecification[0] != nil {
+		mPredefinedScalingMetricSpecification := vPredefinedScalingMetricSpecification[0].(map[string]interface{})
+		if v, ok := mPredefinedScalingMetricSpecification["predefined_scaling_metric_type"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+		if v, ok := mPredefinedScalingMetricSpecification["resource_label"].(string); ok {
+			buf.WriteString(fmt.Sprintf("%s-", v))
+		}
+	}
+	if v, ok := mTargetTrackingConfiguration["scale_in_cooldown"].(int); ok {
+		buf.WriteString(fmt.Sprintf("%d-", v))
+	}
+	if v, ok := mTargetTrackingConfiguration["scale_out_cooldown"].(int); ok {
+		buf.WriteString(fmt.Sprintf("%d-", v))
+	}
+	if v, ok := mTargetTrackingConfiguration["target_value"].(float64); ok {
+		buf.WriteString(fmt.Sprintf("%g-", v))
+	}
+
+	return hashcode.String(buf.String())
+}
+
+// stableMapHash returns a stable hash value for a map.
+func stableMapHash(m map[string]interface{}) int {
```

Comment:
Verify whether this is needed after merge of https://github.com/terraform-providers/terraform-provider-aws/pull/12740, pulling in https://github.com/hashicorp/terraform-plugin-sdk/pull/362.

Reply:
Removed custom set hash functions.